numpyro/handlers.py

```diff
@@ -391,9 +391,18 @@ def process_message(self, msg):
                     else:
                         msg['value'] = self.base_param_map[msg['name']]
         elif self.substitute_fn is not None:
-            value = self.substitute_fn(msg)
-            if value is not None:
-                msg['value'] = value
+            base_value = self.substitute_fn(msg)
+            if base_value is not None:
+                if msg['type'] == 'sample':
+                    msg['value'], msg['intermediates'] = msg['fn'].transform_with_intermediates(
```

Comment:
What happens when the given site is not a `TransformedDistribution`? Then `msg['fn'].trans..` will throw an error, it seems.

Reply:
Maybe we should add a check to that effect, i.e. only do this if `msg['fn']` is a `TransformedDistribution`?

Reply:
If it is not a `TransformedDistribution`, then this will return `base_value ` and empty list. I added it [here](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/distributions/distribution.py#L127) in previous PR for `base_param_map`. :D

Reply:
Oh..interesting, I forgot that the base Distribution class supports this.